src/hooks/useClickedOutside/useClickedOutside.ts

```diff
@@ -0,0 +1,52 @@
+import { useCallback, useMemo } from 'react';
+import { unref, type Unreffable } from '../../utils/unref/unref.js';
+import { useEventListener } from '../useEventListener/useEventListener.js';
+
+/**
+ * Registers a click event listener on the document that triggers a callback function when the
+ * click event occurs outside of the specified element.
+ *
+ * @param ref
+ * @param listener
+ * @param options
+ */
+export function useClickedOutside(
+  target: Unreffable<Element | null>,
+  listener: (mouseEvent: MouseEvent) => void,
+  options?: EventListenerOptions,
+): void {
+  const memoizedOptions = useMemo(
+    () => options,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [...Object.keys(options ?? {}), ...Object.values(options ?? {})],
+  );
+
+  const onClick = useCallback(
+    (event: Event) => {
+      // Using element bounds because a click on a shadow element (e.g. dialog backdrop) is also
```

Comment:
Using this way to check this would return "Inside" if you click on an element that could cover the entire element, so you're not really click the element? I don't think this is what you want.

Reply:
I guess you need to do both then.

1. check if it's inside the expected bounds
2. check if the clicked target is a child of the element

Could you clarify why a backdrop would not influence the element bounding rect?

Also, both variations should have either a story or a test, currently the "backdrop" case doesn't have either?